fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt

```diff
@@ -0,0 +1,235 @@
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.DEFAULT_MAX_LINES
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val ANIMATION_DURATION_MILLIES = 2000
+
+/**
+ * Download indicator.
+ *
+ * @param text The download indicator text to be displayed.
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param contentDescription to be applied to the layout.
+ * @param backgroundColor The background [Color] of the download indicator.
+ * @param textColor [Color] to apply to the download indicator text.
+ * @param icon Optional [Painter] used to display an [Icon] before the download indicator text.
+ * @param iconContentDescription to be applied to the icon.
+ * @param iconModifier to be applied to the icon.
+ * @param iconTint to be applied to the icon.
+ */
+@Composable
+fun DownloadIndicator(
+    text: String,
+    modifier: Modifier = Modifier,
+    contentDescription: String? = null,
+    backgroundColor: Color = FirefoxTheme.colors.actionPrimary,
+    textColor: Color = FirefoxTheme.colors.textActionPrimary,
+    icon: Painter? = null,
+    iconContentDescription: String? = null,
+    iconModifier: Modifier = Modifier,
+    iconTint: Color = FirefoxTheme.colors.iconActionPrimary,
+) {
+    BaseDownloadIndicator(
+        text = text,
+        textColor = textColor,
+        modifier = modifier,
+        buttonContentDescription = contentDescription,
+        icon = icon,
+        backgroundColor = backgroundColor,
+        iconModifier = iconModifier.then(
+            Modifier.wrapContentSize(),
```

Comment:
A bit of a naive question - is this `wrapContentSize()` doing anything for else? I have tried this with and without this and it appears the same in the preview. So, I am wondering what this is doing.

Reply:
Done